src/span.js

```diff
@@ -154,11 +160,12 @@ export default class Span {
      * @param {number} finishTime - The time on which this span finished.
      **/
     finish(finishTime: ?number): void {
-        if (this._duration !== undefined) {
+        if (this.finished) {
             throw new Error('You can only call finish() on a span once.');
         }
 
         if (this._spanContext.isSampled()) {
+            this._finished = true;
```

Comment:
is this right? only if a span is sampled it counts as finished? Shouldn't it be finished if finish is called at least once?